core/textile/client.go

```diff
@@ -102,41 +100,18 @@ func (tc *textileClient) requiresRunning() error {
 }
 
 func (tc *textileClient) GetBaseThreadsContext(ctx context.Context) (context.Context, error) {
-	// TODO: this should be happening in an auth lambda
-	// only needed for hub connections
-	key := os.Getenv("TXL_USER_KEY")
-	secret := os.Getenv("TXL_USER_SECRET")
-
-	if key == "" || secret == "" {
-		return nil, errors.New("Couldn't get Textile key or secret from envs")
-	}
-
 	if ctx == nil {
 		ctx = context.Background()
 	}
-	ctx = common.NewAPIKeyContext(ctx, key)
 
 	var err error
-	var apiSigCtx context.Context
-
-	if apiSigCtx, err = common.CreateAPISigContext(ctx, time.Now().Add(time.Minute), secret); err != nil {
-		return nil, err
-	}
-	ctx = apiSigCtx
 
-	log.Debug("Obtaining user key pair from local store")
-	kc := keychain.New(tc.store)
-	var privateKey crypto.PrivKey
-	if privateKey, _, err = kc.GetStoredKeyPairInLibP2PFormat(); err != nil {
-		return nil, err
-	}
-
-	// TODO: CTX has to be made from session key received from lambda
-	log.Debug("Creating libp2p identity")
-	tok, err := tc.threads.GetToken(ctx, thread.NewLibp2pIdentity(privateKey))
+	log.Debug("Authenticating with Textile Hub")
+	tokStr, err := getHubToken(tc.store, tc.cfg)
 	if err != nil {
-		return nil, err
+		log.Error("Token Challenge Error:", err)
```

Comment:
Shouldn't you return the err here, or is it possible to initialize `thread.Token()` with an empty `tokStr`?

Reply:
Agreed

Reply:
Fixed in 4ef068e